vendor/zkllvm-metacraft-circuits/src/circuits_imp/compute_shuffled_index_imp.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <nil/crypto3/hash/algorithm/hash.hpp>
+#include <nil/crypto3/hash/sha2.hpp>
+
+#include <algorithm>
+#include <array>
+
+#include "compute_shuffled_index_imp.h"
+#include "../circuit_utils/circuit_byte_utils.h"
+#include "../utils/picosha2.h"
+
+using namespace circuit_byte_utils;
+
+uint64_t compute_shuffled_index_imp(uint64_t index, uint64_t index_count, std::array<Byte, 32> seed,
+                                            int SHUFFLE_ROUND_COUNT = 90) {
+    assert_true(index < index_count);
+
+    std::array<Byte, 32 + 1 + 4> source_buffer {};
+
+    //!!! sha256_to_bytes_array(seed, source_buffer);
+    std::copy(seed.begin(), seed.end(), source_buffer.begin());
+
+    // Swap or not (https://link.springer.com/content/pdf/10.1007%2F978-3-642-32009-5_1.pdf)
+    // See the 'generalized domain' algorithm on page 3
+    for (Byte current_round = 0; current_round < SHUFFLE_ROUND_COUNT; current_round++) {
+        source_buffer[32] = current_round;
+
+        //!!! auto eth2digest = hash<hashes::sha2<256>>(source_buffer.begin(), source_buffer.begin() + 33);
+        std::array<Byte, 32> eth2digest_bytes;
```

Comment:
Why is this called eth2digest_bytes?

Reply:
Temporary variable renamed to source_round_hash.